ModulemdTranslationHelpers/Utils.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+import gi
+gi.require_version('Modulemd', '2.0')
+from gi.repository import Modulemd
+
+from babel.messages import Catalog, pofile
+from babel._compat import StringIO, BytesIO
+from datetime import datetime
+
+# Create a Modulemd.ModuleIndex object containing the YAML stream
+fedora_index = Modulemd.ModuleIndex.new()
+fedora_yaml_path = "../modulemd/v2/tests/test_data/f29.yaml"
+ret, failures = fedora_index.update_from_file(fedora_yaml_path, True)
+
+# Retrieve all Modulemd.Module objects
+module_names = fedora_index.get_module_names()
+
+# Create a dictionary where
+# key: stream name
+# value: NSVCA of highest version of that stream
+final_streams = {}
+
+for mod_name in module_names:
+    module = fedora_index.get_module(mod_name)
+
+    # Retrieve all Modulemd.ModuleStreamV2 objects from module
+    streams = module.get_all_streams()
+    for stream in streams:
+        key = stream.get_stream_name()
+        if key not in final_streams.keys():
+            final_streams[key] = stream.get_NSVCA()
+        else:
+            nsvca = stream.get_NSVCA()
+            version = int(nsvca.split(':')[2])
+            if version > int(final_streams[key].split(
+                    ":")[2]):  # found a higher version for same stream
+                final_streams[key] = nsvca
+
+summary_dict = {}
```

Comment:
These should not be separate dictionaries. By the end of this, there must be exactly one dictionary whose keys are the untranslated strings and whose values are a list of places where that string was found.

One example problem: you may have a module and stream that has the same string for both the summary and description fields. What that should look like is:
```
strings_dict["extracted string"] = [ "moduleA:streamA:summary", "moduleA:streamA:description" ]
```